src/engine.py

```
@@ -3136,6 +3136,48 @@ def _append_outbound_custom_vars_to_prompt(self, prompt: str, custom_vars: Dict[
+                campaign_id=getattr(session, 'outbound_campaign_id', None) or "",
+                lead_id=getattr(session, 'outbound_lead_id', None) or "",
+            )
+        except KeyError as e:
+            # Unknown placeholder - leave original text unchanged
+            logger.debug(
+                "Prompt template has unknown placeholder, leaving unchanged",
+                call_id=session.call_id,
+                placeholder=str(e),
+            )
+            return text
+        except Exception as e:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)